### ISSUE TITLE
Failed to Run Android Instance

### ISSUE THREAD

**Issue by RyanJoseph15** [uport-project/uport-mobile#77]
I'm trying to get up and running with an Android instance, following the [README](https://github.com/uport-project/uport-mobile/blob/develop/README.md). I was able to get an iOS instance running without issues. However, `react-native run-android` was giving errors.

1. Clone the project
2. From that directory, in a terminal window, run `native-react run-android`

**Expected behavior**
I would expect an instance of the application to be installed on the adb connected device.

**Actual behavior**
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':react-native-signer-wrapper:debugCompileClasspath'.
> Could not find com.facebook.react:react-native:0.55.3.
  Required by:
      project :react-native-signer-wrapper

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 5s
38 actionable tasks: 1 executed, 37 up-to-date
Could not install the app on the device, read the error above for details.
Make sure you have an Android emulator running or a device connected and have
set up your Android development environment:
https://facebook.github.io/react-native/docs/getting-started.html
```

**Screenshots**
I made sure that I had a device running:
![image](https://user-images.githubusercontent.com/6688146/52965617-7983c080-3362-11e9-855d-3d52cd8a489f.png)

**System:**
 - OS: macOS Mojave
 - Version: 10.14.3

**Smartphone:**
 - Device: Pixel 2XL
 - OS: 5.0 (SDK 21)

**Additional context**
Troubleshooting I've done:
* Using a physical device - this is the actual use-case I'll need this for since I'm working to add bluetooth and location services to add [XYO's BLE SDK](https://developers.xyo.network/docs/en/SDK-Ble-Android/) to pair Proof of Location with Proof of Identity, which was the subject of [the hack](https://github.com/RyanJoseph15/uport-mobile/tree/GeoVerify) my team worked on at [EthDenver](https://www.ethdenver.com/) over the weekend.
* I've also tried running the project from an existing installation of Android Studio, but *"[Open project in Android studio and press play](https://github.com/uport-project/uport-mobile/blob/develop/README.md#run-from-android-studio)"* doesn't seem to to be accurate. When I opened the project, Android Studio did not automatically detect any Run Configurations.
  * A side note on this step, I assumed the correct directory to open was the root directory.
* > Could not find com.facebook.react:react-native:0.55.3.
  * Looking into this from the console output, I wasn't able to resolve the issue after reading through [React-Native](https://facebook.github.io/react-native/docs/getting-started.html)'s documentation.

**Comment by mirceanis** [comment]
There's an extra step needed between cloning the repo and running `react-native run-android`.
The node_modules need to be resolved by running `yarn`.
iOS and Android setups depend on JavaScript environment setup.
This is a bug in the instructions in the readme so thanks for spotting it.

**Comment by mirceanis** [comment]
@RyanJoseph15 I made a PR to fix some of the instructions in the readme (https://github.com/uport-project/uport-mobile/pull/78/files).

TL;DR; 
- clone repo
- run `yarn`
- add firebase config as `android/app/google-services.json`
- run `react-native run-android`

**Comment by RyanJoseph15** [comment]
I'm going through the update README now. Thanks for jumping on this so quickly 😄 

Some additional notes on getting the `google-services.json` file from Firebase:
1. Sign into Firebase with your Google Account.
2. Add Project.
![image](https://user-images.githubusercontent.com/6688146/53028304-7a2c5d80-341b-11e9-8714-39d389eca329.png)
3. Click the Android Icon. We're going to register an Android App.
![image](https://user-images.githubusercontent.com/6688146/53028231-52d59080-341b-11e9-968b-3ed128c366db.png)
4. Add your project and agree to the terms.
![image](https://user-images.githubusercontent.com/6688146/53031318-77cd0200-3421-11e9-8922-1275a96e159e.png)
5. Register your app.
![image](https://user-images.githubusercontent.com/6688146/53028492-d1323280-341b-11e9-8cc6-a0752bdf5107.png)
6. Download the `google-services.json` file and add to your local codebase.
 * Note: `google-services.json` is in the [.gitignore](https://github.com/uport-project/uport-mobile/blob/develop/.gitignore) file. So, nobody will accidentally check in their Firebase files 🎉
![image](https://user-images.githubusercontent.com/6688146/53028506-da230400-341b-11e9-877a-b6e25ab93ed0.png)
7. Add Firebase SDK - don't worry about this step. These are already implemented in the appropriate gradles.
![image](https://user-images.githubusercontent.com/6688146/53032761-57527700-3424-11e9-9f6f-b5185013ed3c.png)
8. Run your app to verify installation: `react-native run-android`, as outlined in the updated README.
![image](https://user-images.githubusercontent.com/6688146/53032815-751fdc00-3424-11e9-9d5c-8f26210424e1.png)

@mirceanis At this point, running the application connects to a running device. However, when using an emulator, it gets stuck on the splash screen. I'm looking into that after attaching `logcat` to the device for errors. I'll update as I find thing.

**Comment by RyanJoseph15** [comment]
@mirceanis, the behavior is the same on a physical device, too.
[Here is the `adb logcat` output](https://github.com/uport-project/uport-mobile/files/2887121/stuck-on-splashscreen-logcat.txt), running on an emulator with the following configurations:
![image](https://user-images.githubusercontent.com/6688146/53134786-6709b280-352d-11e9-92af-88b04c9b8900.png)


Are there additional steps required to get around this?

**Comment by RyanJoseph15** [comment]
I was using an x86_64 image before. Switching to an x86 image resolved being stuck on the splash screen issue.

**Comment by RyanJoseph15** [comment]
Woo-hoo!

![image](https://user-images.githubusercontent.com/6688146/53135713-257b0680-3531-11e9-9ea1-aef7b2fa4d60.png)

**Comment by kanna998** [comment]
Hi Developers,
I'm able to run app using native-react run-android. But after app starts my server which start using "yarn start" is throwing an error as below,
(0/1)C:\Users\<userName>\AndroidStudioProjects\uPort\uport-mobile\node_modules\graceful-fs\polyfills.js:285
        if (cb) cb.apply(this, arguments)
                   ^

TypeError: cb.apply is not a function
    at C:\Users\<userName>\AndroidStudioProjects\uPort\uport-mobile\node_modules\graceful-fs\polyfills.js:285:20    
at FSReqCallback.oncomplete (fs.js:177:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Kindly help me to resolve this issue. 
Thanks in advance
